src/test/java/com/waverleysoftware/BaseTest.java

```diff
@@ -1,16 +1,39 @@
 package com.waverleysoftware;
 
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.ITestContext;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
 
+import java.lang.reflect.Method;
+
+/**
+ * A simple base class for processing tests.
+ */
+@Slf4j
 public class BaseTest {
+
+    protected static final SoftAssertions SOFTASSERT = new SoftAssertions();
+
     @BeforeClass
-    public void setUp() {
-        System.out.println("Setting up on parent level");
+    public void setUp(final ITestContext context) {
+        log.info(context.getCurrentXmlTest().getName());
+        log.info("Starting process on parent level");
+    }
+
+    @BeforeMethod
+    public void setUp(final ITestContext context, final Method method) {
+        log.info(method.getName());
+        log.info(String.valueOf(method.getDeclaredAnnotation(Test.class)));
+        log.info(context.getCurrentXmlTest().getParameter("browser"));
     }
 
     @AfterClass
     public void tearDown() {
-        System.out.println("Ending process on parent level");
+        log.info("Ending process on parent level");
+        SOFTASSERT.assertAll();
```

Comment:
it's not recommended to throw errors within configuration methods... try to move this staff into listener, catch an assertion error and manually change test result to failure...